Tests/UnitTests/SubscriberAttributes/SubscriberAttributeTests.swift

```diff
@@ -220,6 +220,53 @@ class SubscriberAttributeTests: TestCase {
         expect(dict1.individualizedCacheKeyPart) != dict2.individualizedCacheKeyPart
     }
 
+    func testDictionaryIndividualizedCacheKeyPartIsDeterministicRegardlessOfInsertionOrder() {
+        let dateProvider = MockDateProvider(stubbedNow: Date(timeIntervalSince1970: 1000))
```

Comment:
by nature of the bug being non-deterministic, this test would have failed some percentage of the time in the old implementation. but I did see it fail! and it succeeds with the current one